libraries/AP_AIS/AP_AIS.cpp

```diff
@@ -0,0 +1,764 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Automatic Identification System, https://gpsd.gitlab.io/gpsd/AIVDM.html
+
+// ToDo: enable receiving of the Mavlink AIS message, type bitmask?
+
+#include "AP_AIS.h"
+
+#if HAL_AIS_ENABLED
+
+#include <AP_Logger/AP_Logger.h>
+#include <GCS_MAVLink/GCS_MAVLink.h>
+#include <GCS_MAVLink/GCS.h>
+
+#define LIST_SIZE_DEFAULT 25
+#define TIME_OUT_DEFAULT 600
```

Comment:
Not a blocker, but these are very generic macro names with no explanation of what they do.

I'd really rather see the numbers just put in-line - if someone is tuning them they're going to go and look at the context they're used in anyway, so may as well not split the code up.